homeassistant/components/squeezebox/media_player.py

```diff
@@ -182,133 +180,52 @@ async def async_service_handler(service):
-            return False
-
-        try:
-            return data["result"]
-        except AttributeError:
-            _LOGGER.error("Received invalid response: %s", data)
-            return False
-
-
 class SqueezeBoxDevice(MediaPlayerEntity):
-    """Representation of a SqueezeBox device."""
+    """
+    Representation of a SqueezeBox device.
 
-    def __init__(self, lms, player_id, name):
+    Wraps a pysqueezebox.Player() object.
+    """
+
+    def __init__(self, hass, player):
         """Initialize the SqueezeBox device."""
         super().__init__()
-        self._lms = lms
-        self._id = player_id
-        self._status = {}
-        self._name = name
+        self._hass = hass
+        self._player = player
         self._last_update = None
-        _LOGGER.debug("Creating SqueezeBox object: %s, %s", name, player_id)
+        _LOGGER.debug(
+            "Creating SqueezeBox object: %s, %s", player.name, player.player_id
+        )
 
     @property
     def name(self):
         """Return the name of the device."""
-        return self._name
+        return self._player.name
 
     @property
     def unique_id(self):
         """Return a unique ID."""
-        return self._id
+        return self._player.player_id
 
     @property
     def state(self):
         """Return the state of the device."""
-        if "power" in self._status and self._status["power"] == 0:
-            return STATE_OFF
-        if "mode" in self._status:
-            if self._status["mode"] == "pause":
+        if self._player.power is not None:
+            if not self._player.power:
+                return STATE_OFF
+        if self._player.mode:
+            if self._player.mode == "pause":
                 return STATE_PAUSED
-            if self._status["mode"] == "play":
+            if self._player.mode == "play":
                 return STATE_PLAYING
-            if self._status["mode"] == "stop":
+            if self._player.mode == "stop":
                 return STATE_IDLE
         return None
 
-    async def async_query(self, *parameters):
-        """Send a command to the LMS."""
-        return await self._lms.async_query(*parameters, player=self._id)
-
     async def async_update(self):
-        """Retrieve the current state of the player."""
-        tags = "adKl"
-        response = await self.async_query("status", "-", "1", f"tags:{tags}")
-
-        if response is False:
-            return
-
+        """Update the Player() object."""
+        await self._player.async_update()
         last_media_position = self.media_position
-
-        self._status = {}
-
-        try:
-            self._status.update(response["playlist_loop"][0])
-        except KeyError:
-            pass
-        try:
-            self._status.update(response["remoteMeta"])
-        except KeyError:
-            pass
-
-        self._status.update(response)
-
         if self.media_position != last_media_position:
```

Comment:
Why do we do this check? Right above we set `last_media_position` to `self.media_position`.

Reply:
Thanks for catching this. That was a mistake. I inadvertently reversed these operations when refactoring. It should capture last_media_position before running player.async_update(). Fixed in new commit.